src/components/map/MapLibre.tsx

```diff
@@ -209,6 +222,7 @@ export const MapLibre = ({
         rotateEnabled={false}
         style={[styles.map, mapStyle]}
         zoomEnabled
+        onPress={!!onMapPress && handleMapPress}
```

Comment:
this should not depend on `onMapPress`. we need some other flag to indicate, that setting a pin is allowed.
and if this is activated, other map press actions should not be applied i guess. imagine a map press would navigate somewhere, you would not want this to happen after setting a new pin.

Reply:
I added this control here because the `handleMapPress` function is useless without the `onMapPress` function. We need the changes made inside `onMapPress`. Also, if you look at the `handleMapPress` function, you can see that the `onMapPress` function is inside it. But I think that with the controls made in `handleMapPress`, we prevent the `onMapPress` control.

Reply:
ok, in this case it should be handled inside `handleMapPress` and not when passing the prop.

Reply:
i will add the flag `setPinEnabled` to determine the case